core/src/main/java/com/linecorp/armeria/server/ServiceNaming.java

```diff
@@ -72,6 +73,18 @@ static ServiceNaming simpleTypeName() {
         };
     }
 
+    /**
+     * Returns the {@link ServiceNaming} that returns the shortened service name from the full name of an RPC
+     * stub class or the innermost class from the given service. It follows Logback's abbreviation algorithm.
+     * Please note that the rightmost segment in a service name is never abbreviated. For instance,
+     * {@code com.foo.bar.HelloService} is able to be shorten to {@code c.f.b.HelloService}.
+     *
+     * @see <a href="http://logback.qos.ch/manual/layouts.html">Logback's abbreviation algorithm</a>
+     */
+    static ServiceNaming shorten(int targetLength) {
```

Comment:
What's the sensible default for `targetLength`? Could we add `shorten()` without a parameter that returns the abbreviated naming that abbreviates everything but the last component?

Reply:
> Could we add `shorten()` without a parameter that returns the abbreviated naming that abbreviates everything but the last component?

Absolutely. We are able to use `0` as a default value. I will add `shorten()` by using the default value.

Reply:
Sounds good!